native/mod_manager/mod_manager.c

```diff
@@ -2980,17 +3012,26 @@ static char *process_domain(request_rec *r, char **ptr, int *errtype, const char
 }
 
 /* XXX: move to mod_proxy_cluster as a provider ? */
-static void printproxy_stat(request_rec *r, int reduce_display, const proxy_worker_shared *proxystat)
+static void printproxy_stat(request_rec *r, int reduce_display, nodeinfo_t *node)
 {
-    char *status = NULL;
-    status = proxystat->status & PROXY_WORKER_NOT_USABLE_BITMAP ? "NOTOK" : "OK";
+    char *status;
+    proxy_worker_shared *proxystat = read_shared_by_node(r, node);
+    if (!proxystat) {
+        ap_log_error(APLOG_MARK, APLOG_ERR, 0, r->server, "printproxy_stat can't find worker");
+        status = "NOTOK";
+    } else
+        status = proxystat->status & PROXY_WORKER_NOT_USABLE_BITMAP ? "NOTOK" : "OK";
+     
 
     if (reduce_display) {
         ap_rprintf(r, " %s ", status);
     } else {
-        ap_rprintf(r, ",Status: %s,Elected: %d,Read: %d,Transferred: %d,Connected: %d,Load: %d", status,
-                   (int)proxystat->elected, (int)proxystat->read, (int)proxystat->transferred, (int)proxystat->busy,
-                   proxystat->lbfactor);
+        if (!proxystat)
+            ap_rprintf(r, ",Status: NOTOK,Elected: 0,Read: 0,Transferred: 0,Connected: 0,Load: 0");
+        else
+            ap_rprintf(r, ",Status: %s,Elected: %d,Read: %d,Transferred: %d,Connected: %d,Load: %d", status,
+                       (int)proxystat->elected, (int)proxystat->read, (int)proxystat->transferred, (int)proxystat->busy,
+                       proxystat->lbfactor);
     }
```

Comment:
This does not have to be nested, `else if (!proxystat) { ... } else { ... }` should be ok.

Reply:
proxystat = apr_pcalloc(r->pool, sizeof(proxy_worker_shared));
would be better no? we are already testing proxystat to set status.

Reply:
I don't understand. The `proxystat` is from `read_shared_by_node`, isn't it?

Reply:
yes.

Reply:
Ok, now I get it. But this solution has a problem that in case status is NOTOK, then nobody can rely on the rest of the data/information, because it might be 1) valid data, but worker is not usable; 2) the data are "random" because it's just freshly allocated memory. And there is no way how to say which one it is, or am I missing something?

Reply:
it is 1) the worker is NOT yet available in fact.
I first put an assert() here but with mod_lbmethod_cluster/mod_proxy_balancer it happens when starting tomcat...

Reply:
Anyway the status NOTOK is correct ;-) you can't use the worker.

Reply:
Ok, but wouldn't it be better to have this instead?

```c
if (reduce_display || strcmp(status, "NOTOK") == 0) {
        ap_rprintf(r, " %s ", status);
} else {
        ap_rprintf(r, ",Status: %s,Elected: %d,Read: %d,Transferred: %d,Connected: %d,Load: %d", status,
                   (int)proxystat->elected, (int)proxystat->read, (int)proxystat->transferred, (int)proxystat->busy,
                   proxystat->lbfactor);
}
```

So that we do not print values that are unusable.

Reply:
well that might break clients parsing the result...

Reply:
Ok, that did not occur to me.